src/basic/selinux-util.c

```diff
@@ -84,24 +84,33 @@ void mac_selinux_retest(void) {
 }
 
 #if HAVE_SELINUX
-#  if HAVE_MALLINFO
-static struct mallinfo mallinfo_nowarn(void) {
-        /* glibc has deprecated mallinfo(), but the replacement malloc_info() returns an XML blob ;=[ */
+#  if HAVE_MALLINFO2
+#    define HAVE_GENERIC_MALLINFO 1
+typedef struct mallinfo2 generic_mallinfo;
+static generic_mallinfo generic_mallinfo_get(void) {
+        return mallinfo2();
+}
+#  elif HAVE_MALLINFO > 1
```

Comment:
Hm? Why `> 1`?

Reply:
oops, it's left-over when i tested the other codepaths here locally, and of course i accidentally left it in. will update shortly.